sdk/containerregistry/azure-containers-containerregistry/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,127 @@
+    .buildClient()
+    .getRepository(repositoryName);
+try {
+    containerRepository.getProperties();
+} catch (HttpResponseException exception) {
+    // Do something with the exception.
+}
+```
+
+Azure Container Registry client library log errors by default. The errors and warnings in the logs or traces generally provide
+useful insights into what went wrong and sometimes include corrective actions to fix issues.
+
+Make sure logging or tracing are enabled to monitor the behavior of the application and to troubleshoot issues with Azure Container Registry library. 
+
+### Enable client logging
+
+The Azure client libraries for Java have two logging options:
+
+* A built-in logging framework.
+* Support for logging using the [SLF4J](https://www.slf4j.org/) interface.
+
+Refer to the instructions in this reference document on how to [configure logging in Azure SDK for Java](https://docs.microsoft.com/azure/developer/java/sdk/logging-overview).
+
+### Enable HTTP request/response logging
+
+Reviewing the HTTP request sent or response received over the wire to/from the Azure Container Registry service can be useful in
+troubleshooting issues. To enable logging the HTTP request and response payload, the `ContainerRegistryClient` can be configured as shown below:
+
+```java readme-sample-enablehttplogging
+ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(defaultCredential)
+    .httpLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+    .buildClient();
+```
+
+Alternatively, you can configure logging HTTP requests and responses for your entire application by setting the
+following environment variable. Note that this change will enable logging for every Azure client that supports logging
+HTTP request/response.
+
+Environment variable name: `AZURE_HTTP_LOG_DETAIL_LEVEL`
+
+| Value            | Logging level                                                        |
+|------------------|----------------------------------------------------------------------|
+| none             | HTTP request/response logging is disabled                            |
+| basic            | Logs only URLs, HTTP methods, and time to finish the request.        |
+| headers          | Logs everything in BASIC, plus all the request and response headers. |
+| body             | Logs everything in BASIC, plus all the request and response body.    |
+| body_and_headers | Logs everything in HEADERS and BODY.                                 |
+
+**NOTE**: When logging the body of request and response, please ensure that they do not contain confidential
+information. When logging headers, the client library has a default set of headers that are considered safe to log
+but this set can be updated by updating the log options in the builder as shown below:
+
+```java
+clientBuilder.httpLogOptions(new HttpLogOptions().addAllowedHeaderName("safe-to-log-header-name"))
```

Comment:
Does the Container Registry SDK provide a way to get pre-vetted headers and query parameters that are safe to log? Where a customer could append additional allowed headers or query parameters.

Reply:
this example would include all the common headers across all SDKs.
there are no ACR-specific headers that SDK makes use of, however, services supports a few custom ones on response. If SDK will define any ACR specific, by all means, it would be great to add them by default into allowed headers.

BTW, do we have a precedent where we allow headers/query params specific to SDK?

Reply:
Yeah, Storage and Search have it.

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/BlobClientBuilder.java#L500

https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java#L290

But this is something that can be followed up on after this PR.